infrastructure/http_metadata_service_test.go

```diff
@@ -422,4 +422,66 @@ func describeHTTPMetadataService() {
 		})
 
 	})
+
+	Describe("Decode a string of user data which has been encoded using base-64 encoding by IaaS", func() {
+		var (
+			ts *httptest.Server
+		)
+
+		createHandlerFunc := func(count int) func(http.ResponseWriter, *http.Request) {
+			initialCount := 0
```

Comment:
i dont think we need to test retry logic in this set of tests. i would simplify the test to be more like: https://github.com/jianqiu/bosh-agent/blob/943e43a2e5dbc9771c81b32bc4e00fe95e2b50d4/infrastructure/http_metadata_service_test.go#L187